cudapoa/samples/sample_cudapoa.cpp

```diff
@@ -190,113 +173,130 @@ int main(int argc, char** argv)
         std::exit(0);
     }
 
-    // Load input data. Each POA group is represented as a vector of strings. The sample
-    // data for short reads has many such POA groups to process, hence the data is loaded into a vector
-    // of vector of strings. Long read sample creates one POA group.
+    // Load input data. Each window is represented as a vector of strings. The sample
+    // data has many such windows to process, hence the data is loaded into a vector
+    // of vector of strings.
     std::vector<std::vector<std::string>> windows;
 
-    // Define upper limits for sequence size, graph size ....
-    BatchSize batch_size;
-
     if (long_read)
     {
         const std::string input_file = std::string(CUDAPOA_BENCHMARK_DATA_DIR) + "/sample-bonito.txt";
-        generate_window_data(input_file, -1, 6, windows, batch_size);
+        parse_window_data_file(windows, input_file, -1);
     }
     else
     {
         const std::string input_file = std::string(CUDAPOA_BENCHMARK_DATA_DIR) + "/sample-windows.txt";
-        generate_window_data(input_file, 1000, 100, windows, batch_size);
+        parse_window_data_file(windows, input_file, 1000);
     }
 
-    // Initialize batch.
-    std::unique_ptr<Batch> batch = initialize_batch(msa, banded, batch_size);
-
-    // Loop over all the POA groups, add them to the batch and process them.
-    int32_t window_count = 0;
-
-    for (int32_t i = 0; i < get_size(windows);)
+    // Create a vector of POA groups based on windows
+    std::vector<Group> poa_groups(windows.size());
+    for (int32_t i = 0; i < get_size(windows); ++i)
     {
-        const std::vector<std::string>& window = windows[i];
-
-        Group poa_group;
+        Group& group = poa_groups[i];
         // Create a new entry for each sequence and add to the group.
-        for (const auto& seq : window)
+        for (const auto& seq : windows[i])
         {
             Entry poa_entry{};
             poa_entry.seq     = seq.c_str();
             poa_entry.length  = seq.length();
             poa_entry.weights = nullptr;
-            poa_group.push_back(poa_entry);
+            group.push_back(poa_entry);
         }
+    }
 
-        std::vector<StatusType> seq_status;
-        StatusType status = batch->add_poa_group(seq_status, poa_group);
+    // analyze the POA groups and create a minimal set of batches to process them all
+    std::vector<BatchSize> list_of_batch_sizes;
+    std::vector<std::vector<int32_t>> list_of_groups_per_batch;
 
-        // NOTE: If number of windows smaller than batch capacity, then run POA generation
-        // once last window is added to batch.
-        if (status == StatusType::exceeded_maximum_poas || (i == get_size(windows) - 1))
+    cudapoa::get_multi_batch_sizes(list_of_batch_sizes, list_of_groups_per_batch, poa_groups, banded, msa);
```

Comment:
since we're using the `cudapoa` namespace, we don't need to explicitly add `cudapoa::get_multi_batch_sizes`. It can just be `get_multi_batch_sizes`

Reply:
that's right, fixed.